osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/infra/jms/messageprocessors/DomainResponseMessageProcessor.java

```diff
@@ -80,28 +84,31 @@ public void processMessage(final ObjectMessage message) {
 
             final CorrelationIds ids = new CorrelationIds(organisationIdentification, deviceIdentification,
                     correlationUid);
-            this.handleMessage(ids, messageType, resultType, resultDescription, dataObject);
 
+            LOGGER.info("Handle message of type {} for device {} with result: type {}, description {}.", messageType,
+                    ids.getDeviceIdentification(), resultType, resultDescription);
+            this.handleMessage(messageType, dataObject);
         } catch (final RuntimeException e) {
             handleError(e, correlationUid);
         }
     }
 
-    private void handleMessage(final CorrelationIds ids, final String messageType,
-            final ResponseMessageResultType resultType, final String resultDescription, final ResponseMessage message) {
-        LOGGER.debug("Handling message of type {} for device {} with result: type {}, description {}.", messageType,
-                ids.getDeviceIdentification(), resultType, resultDescription);
+    private void handleMessage(final String messageType, final ResponseMessage message) {
 
         final Serializable dataObject = message.getDataObject();
-        if (!(dataObject instanceof MeasurementReport)) {
-            LOGGER.error("For this component we only handle measurement reports");
-        } else {
+        if (dataObject instanceof MeasurementReport) {
             this.meterReadingProducer.send(this.mapper.map(dataObject, MeasurementReport.class));
+        } else if (dataObject instanceof String && "GET_DATA".equals(messageType)) {
+            this.peakShavingProducer.send((String) dataObject);
+        } else {
+            LOGGER.warn(
+                    "For this component we only handle measurement reports and MQTT GET_DATA responses. Received message type: {}, message {}",
+                    messageType, dataObject);
         }
+
```

Comment:
The MeasurementReport processing was only used for the hackathon. I have removed that code. We now only process the GET_DATA for peak shaving. Because we don't handle anything else yet, I'm leaving out the "message processor per message type" stuff, which we have elsewhere. When we get additional messages to handle, we will make it more generic. An if - else if - else if - else if... construction should be avoided.